core/hub/sync.go

```diff
@@ -35,31 +30,32 @@ func CheckValidMode(mode string) error {
 
 // QueryHub queries the BTFS-Hub to retrieve the latest list of hosts info
 // according to a certain mode.
-func QueryHub(node *core.IpfsNode, mode string) ([]*info.Node, error) {
-	config, err := node.Repo.Config()
-	if err != nil {
-		return nil, err
-	}
-
-	hubUrl := config.Services.HubDomain
-
-	params := "?id=" + node.Identity.Pretty()
+func QueryHub(node *core.IpfsNode, mode string) ([]*hubpb.Host, error) {
 	switch mode {
 	case HubModeScore:
 		// Already the default on hub api
 	default:
 		return nil, fmt.Errorf(`Mode "%s" is not yet supported`, mode)
 	}
 
-	resp, err := http.Get(hubUrl + params)
+	config, err := node.Repo.Config()
 	if err != nil {
-		return nil, fmt.Errorf("Failed to query BTFS-Hub service: %v", err)
-	}
-
-	var hq hostsQuery
-	if err := json.NewDecoder(resp.Body).Decode(&hq); err != nil {
 		return nil, err
 	}
+	var resp *hubpb.HostsResp
+	err = grpc.HubQueryClient(config.Services.HubDomain).WithContext(context.Background(), func(ctx context.Context,
```

Comment:
We have ctx now, so can you make `QueryHub` take ctx from caller and pass it down

Reply:
sure, that's the right way.